src/Keboola/StorageApi/Client.php

```diff
@@ -171,11 +167,14 @@ private function initClient()
         ]);
 
         $handlerStack->push((RequestTimeoutMiddleware::factory()));
-        $handlerStack->push(Middleware::log(
+
+        // this needs to run after http_errors resolves so the error has proper log level
+        $handlerStack->before('http_errors', Middleware::log(
             $this->logger,
-            new MessageFormatter('{hostname} {req_header_User-Agent} - [{ts}] "{method} {resource} {protocol}/{version}" {code} {res_header_Content-Length}'),
+            new LogMessageFormatter(),
             LogLevel::DEBUG
         ));
```

Comment:
nestacilo by pouzit `BodySummarizer`, ktery resi prave to truncatovani?
```php
    private function initClient()
    {
        $handlerStack = HandlerStack::create([
            'backoffMaxTries' => $this->backoffMaxTries,
        ]);

        $handlerStack->push(Middleware::httpErrors(
            new BodySummarizer(1000) // <-- truncate to 1000 characters
        ));

        $handlerStack->push((RequestTimeoutMiddleware::factory()));
        $handlerStack->push(Middleware::log(
            $this->logger,
            new MessageFormatter('{hostname} {req_header_User-Agent} - [{ts}] "{method} {resource} {protocol}/{version}" {code} {res_header_Content-Length}'),
            LogLevel::DEBUG
        ));
        $this->client = new \GuzzleHttp\Client([
            'base_uri' => $this->apiUrl,
            'handler' => $handlerStack,
        ]);
    }
```

Reply:
Teoreticky to stacilo, pokud by si jeste pred tim ten `http_errors` middleware odregistroval (protoze `HandlerStack::create` ho zaregistruje s defaultni konfiguraci).

```
        $handlerStack = HandlerStack::create([
            'backoffMaxTries' => $this->backoffMaxTries,
        ]);

        $handlerStack->remove('http_errors'); // <-- odregistrovat defaultni `http_errors`
        $handlerStack->unshift(Middleware::httpErrors(
            new BodySummarizer(1000) // <-- truncate to 1000 characters
        ), 'http_errors'); // <-- zaregistrovat custom `http_errors` pres `unshift` na zacatek stacku, aby byl na stejne pozici jako puvodni
```

Jako je to o kus min kodu :slightly_smiling_face:, nic extra slozityho, jen uz mi to prave prislo takovy vosajslicht. Dalsi za me vyhoda (ale to je na posouzeni) mit to v loggeru je IMHO, ze kdyz tu vyjimku nekde vys zachytit/ohandlujes, nikde to nemas. Kdezto ten logger to propise do PT bez ohledu na to, jestli vystrelenou vyjimku dal nejak handlujes.